addie/rietveld/braggtree.py

```diff
@@ -236,6 +236,34 @@ def do_plot_ws(self):
         item_list = [str(item.text()) for item in item_list]
         item_list.sort()
 
+        # ZYP -> If both parent and children are selected from the tree, we want
+        # to get rid of children from the selected wks list. Meanwhile, we want
+        # to check the box corresponding to the selected children.
+        leaf_dict_temp = self._main_window.rietveld_ui.treeWidget_braggWSList._leafDict
+        for item_temp in item_list:
+            if item_temp not in list(leaf_dict_temp.keys()):
+                parent_found = False
+                for key, list_temp in leaf_dict_temp.items():
+                    for item_temp_1 in list_temp:
+                        if item_temp in item_temp_1:
+                            bank_temp = list_temp.index(item_temp_1) + 1
+                            self._main_window._braggBankWidgets[bank_temp].setChecked(True)
+                            parent_found = True
+                            parent_in_tree = key
+                            break
+                    if parent_found:
+                        break
```

Comment:
Looks like this is repeated below in lines 350-366 (correct me if wrong)
From DRY (Don't Repeat Yourself) principle, maybe we should move this to a function / class method and reuse that in both spots.
Thoughts?

Reply:
> Looks like this is repeated below in lines 350-366 (correct me if wrong)
> From DRY (Don't Repeat Yourself) principle, maybe we should move this to a function / class method and reuse that in both spots.
> Thoughts?

Yes. Indeed there is a duplicate piece of these codes here. Will create a function.